text/0000-default-field-values.md

```diff
@@ -0,0 +1,1979 @@
+
+In particular, `Foo { .. }` as sugar slightly improves ergonomics.
+However, it has some notable problems:
+
++ Because it desugars to `Foo { ..Default::default() }`, it cannot be required
+  that the expression is a constant one. This carries all the problems noted in
+  the previous section on why default field values should be a `const` context.
+
++ There is no way of implementing a `Default` implementation that has mandatory
+  fields for users to specify during value construction.
+
++ It provides zero improvements to the ergonomics of *specifying* defaults,
+  only for using them. Arguably, the most important aspect of this RFC is
+  not the syntax `Foo { .. }` but rather the ability to provide default values
+  for field.
+
++ By extension, the improvement to documentation clarity is lost.
+
++ The trait `Default` must now become a `#[lang_item]`. This is a sign of
+  increasing the overall magic in the system; meanwhile, this proposal makes
+  the default values provided usable by other custom derive macros.
+
+Thus in conclusion, while desugaring `..` to `Default::default()` has lower cost,
+it also provides significantly less value to the point of not being worth it.
+
+## `..` is useful as a marker
+
+One possible change to the current design is to permit filling in defaults
+by simply writing `Foo {}`; in other words, `..` is simply dropped from the
+expression.
+
+Among the benefits are:
+
++ To enhance ergonomics of initialization further.
+
++ To introduce less syntax.
+
++ To be more in line with how other languages treat default values.
+
+Among the drawbacks are:
+
++ The syntax `Foo { .. }` is no longer introduced to complement the identical
+  pattern syntax. As aforementioned, destruction (and pattern matching)
+  generally attempts to follow construction in Rust. Because of that,
+  introducing `Foo { .. }` is essentially cost-free in terms of the complexity
+  budget. It is arguably even cost-negative.
+
++ By writing `Foo { .. }`, you have there is explicit indication that default
+  values are being used; this enhances local reasoning further.
+
+This RFC requires the `..` to get defaulted fields because it wants to continue
+to allow the workflow of intentionally *not* including `..` in the struct
+literal expression so that when a user adds a field they get compilation errors
+on every use -- just like is currently possible in patterns by not including
+`..` in the struct pattern.
+
+## Named function arguments with default values
+
+A frequently requested feature is named function arguments. Today, the way to
+design around the lack of these in the language are:
+
++ Builder pattern
++ Defining a `struct` "bag-object" where optional fields are set, making users
+  call functions in the following way:
+  `foo(mandatory, Optionals { bar: 42, ..Default::default() })`
++ Provide multiple methods: `fn foo(mandatory)` *and* `fn foo_with_bar(mandatory, bar)`
```

Comment:
I expected this section would draw a connection to the RFC, but it just ends here and leaves that as an exercise to the reader. I guess the point is that the second item in the list can now be written more concisely?